### DISCUSSION TITLE
Using Roles on two models

### DISCUSSION THREAD

**Discussion by dah-HUB** [spatie/laravel-permission#2270] [Q&A] [answered]
Hi there,
In an app, I'm using this great package to manage roles and permissions of the User model.

I have another model 'Customer' and I want to also manage only its roles, Ex : Customer of this role is affected to that user,  Customer of that role can receive mails,......

So I have this 'users_model_has_roles'  table :

role_id | model_type                  | model_id
2          | App\Models\User         | 1
2          | App\Models\User         | 2
2          | App\Models\Customer | 1
2          | App\Models\Customer | 2
2          | App\Models\User         | 3

when I do : User::first()->roles and Customer::first()->roles all the results are good
But,
when I do : \Spatie\Permission\Models\Role::find(2)->users
I retrieve 3 results (all from the model User), 
It's normal, but the question is how to retrieve the other Customers starting from the Role model ?

Thanks in advance

**Comment by parallels999** [comment, marked as answer]
The model comes from guard, you could change config default guard on the fly for Customers query
